contrib/net/http/http.go

```diff
@@ -13,13 +13,13 @@ import (
 // ServeMux is an HTTP request multiplexer that traces all the incoming requests.
 type ServeMux struct {
 	*http.ServeMux
-	cfg *muxConfig
+	cfg *config
 }
 
 // NewServeMux allocates and returns an http.ServeMux augmented with the
 // global tracer.
-func NewServeMux(opts ...MuxOption) *ServeMux {
-	cfg := new(muxConfig)
+func NewServeMux(opts ...Option) *ServeMux {
```

Comment:
`ServeMux` should also benefit from the new option, let's just change [line 41](https://github.com/DataDog/dd-trace-go/blob/7515d751c02467b60c9044140db3b7cace04c44f/contrib/net/http/http.go#L41) to be:
```go
opts := mux.cfg.spanOpts
```